### PR TITLE
WiP: compatibility matrix update Q2 2022

### PR DESCRIPTION
**Pull request by macolo**
## Description

This is about introducing django-cms 3.10 and 3.11 to the [compatibility table](https://docs.django-cms.org/en/develop/#software-version-requirements-and-release-notes).

## Related resources

* https://github.com/django-cms/django-cms/issues/7312

## Checklist

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [ ] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -157,24 +157,27 @@ by a long-term support policy.
...
-
-
-.. _Python: https://www.python.org
-.. _Django: https://www.djangoproject.com
+===========  ==== === === === === === ===   === === === === === === === ====
+django CMS   Python                         Django
+-----------  ----------------------------   --------------------------------
+\            3.10 3.9 3.8 3.7 3.6 3.5 3.4   4.0 3.2 3.1 3.0 2.2 2.1 2.0 1.11
+===========  ==== === === === === === ===   === === === === === === === ====
+4.2.x        ✓    ✓   ✓   ×   ×   ×   ×     ✓   ✓   ×   ×   ×   ×   ×   ×
```
Comment (Aiky30) [inline docs/index.rst:165]: @macolo There is no 4.2.x unless there are plans that I am not aware of. 
```suggestion
```

File: docs/index.rst [hunk 2]
```diff
@@ -157,24 +157,27 @@ by a long-term support policy.
...
+4.0.x        ✓    ✓   ✓   ✓   ×   ×   ×     ✓   ✓   ×   ×   ×   ×   ×   ×
```
Comment (Aiky30) [inline docs/index.rst:166]: My apologies for the misinformation regarding 3.11 having the same conditions as 4.0, I forgot about the newer python 3.10 and django 4.0 changes. The v4 branches aren't yet structured in a way that we can start adding those changes. 

```suggestion
4.0.x      ×    ✓   ✓   ✓   ×   ×   ×     ×   ✓   ✓   ✓   ✓   ×   ×   ×
```

**Review by Aiky30 — CHANGES_REQUESTED** [review]
Some changes needed for the V4 compatibility.

**Review by Aiky30 — APPROVED** [review]
django-cms v4 is correct @macolo. :-)

**Comment by marksweb** [comment]
Looks good @macolo 

If you change the PR title suffix from "WiP:" to "docs:" it'll pass that failing check.

**Review by marksweb — APPROVED** [review]

**Comment by fsbraun** [comment]
I took the liberty to update the matrix according to the current state and include Django 4.1 on the list. @marksweb can you re-review this?

**Comment by fsbraun** [comment]
Updated with release 3.11.1 :-)

**Pull request closed without merging**